port/linux/omrvmem.c

```diff
@@ -964,49 +965,75 @@ default_pageSize_reserve_memory(struct OMRPortLibrary *portLibrary, void *addres
 	int flags = MAP_PRIVATE;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
+	BOOLEAN useBackingFile = FALSE;
 
 	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
 
+	if(mode & OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE) {
+		useBackingFile = TRUE;
+	} else {
 #if defined(MAP_ANONYMOUS)
-	flags |= MAP_ANONYMOUS;
+		flags |= MAP_ANONYMOUS;
+		useBackingFile = FALSE;
 #elif defined(MAP_ANON)
-	flags |= MAP_ANON;
+		flags |= MAP_ANON;
+		useBackingFile = FALSE;
 #else
-	fd = portLibrary->file_open(portLibrary, "/dev/zero", EsOpenRead | EsOpenWrite, 0);
-	if (-1 != fd)
+		useBackingFile = TRUE;
 #endif
-	{
-		if (0 != (OMRPORT_VMEM_MEMORY_MODE_COMMIT & mode)) {
-			protectionFlags = get_protectionBits(mode);
-		} else {
-			flags |= MAP_NORESERVE;
+	}
+
+	char filename[FILE_NAME_SIZE];
```

Comment:
Did you try using `memfd_create()` instead of opening a named file?  I can't exactly tell what versions of Linux support this call. http://man7.org/linux/man-pages/man2/memfd_create.2.html

Reply:
I haven't, I'll try that and test to see if it works.